src/main/java/com/adyen/Client.java

```diff
@@ -77,6 +81,33 @@ public Client(String username, String password, Environment environment, String
         this(username, password, environment, null, applicationName);
     }
 
+    /**
+     * Use this constructor to create client for client certificate authentication along with API key.
+     * Note: Client certificate authentication is only applicable for PAL and Checkout services in LIVE,
+     * Other services will just use API key for authentication.
+     * @param trustStore Trust store containing server certificate
+     * @param clientKeyStore Client Key store containing client certificate and key
+     * @param clientKeyStorePassword Password for client key store
+     * @param apiKey Adyen API Key
+     * @param region Data center region (EU/US/AU), default EU if not provided
+     */
+    public Client(KeyStore trustStore, KeyStore clientKeyStore, String clientKeyStorePassword, String apiKey, Region region) {
+        this();
```

Comment:
minor: this(String apiKey, Environment environment) can be used here.

Reply:
thanks, will update it,

Reply:
`this(String apiKey, Environment environment)` sets also the other endpoints(marketpay, hpp,...) that do not support client certificate auth.
Do we really want the merchants to use the same client with the certificate, also for "non certificate" endpoints?

Reply:
`this(apiKey, Environment.LIVE)` was just to replace individual `this(),         this.setEnvironment(Environment.LIVE, null); and this.config.setApiKey(apiKey);` so the behaviour is same.

However if we don't want to setup other endpoints then i can remove the `this.setEnvironment(Environment.LIVE, null)` and just set PAL and Checkout endpoints.